google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java

```diff
@@ -260,6 +262,84 @@ public void downloadTo(Path path) {
     downloadTo(path, new BlobSourceOption[0]);
   }
 
+  /**
+   * Uploads the given file path to this blob using specified blob write options.
+   *
+   * @param path file to be uploaded
+   * @param options blob write options
+   * @return updated blob
+   * @throws StorageException upon failure
+   */
+  public Blob uploadFrom(Path path, BlobWriteOption... options) {
+    if (!Files.exists(path)) {
+      throw new StorageException(0, path + ": No such file");
+    }
+    if (Files.isDirectory(path)) {
+      throw new StorageException(0, path + ": Is a directory");
+    }
+    try (InputStream input = Files.newInputStream(path)) {
+      return uploadFrom(input, options);
+    } catch (IOException e) {
+      throw new StorageException(e);
+    }
+  }
+
+  /**
+   * Uploads the given content to this blob using specified blob write options.
+   *
+   * @param input content to be uploaded
+   * @param options blob write options
+   * @return updated blob
+   * @throws StorageException upon failure
+   */
+  public Blob uploadFrom(InputStream input, BlobWriteOption... options) {
+    try (WriteChannel writer = storage.writer(this, options)) {
+      upload(input, writer);
+    } catch (IOException e) {
+      throw new StorageException(e);
+    }
+    BlobId blobId = getBlobId();
+    return storage.get(BlobId.of(blobId.getBucket(), blobId.getName()));
+  }
+
+  static void upload(InputStream input, WriteChannel writer) {
+    upload(input, writer, DEFAULT_CHUNK_SIZE);
+  }
+
+  /**
+   * Uploads the given content to the storage using specified write channel and the given
+   * buffer size. The default buffer size is 15 MB. Larger buffer sizes may improve the upload
+   * performance but require more memory. It could cause OutOfMemoryError or add significant garbage
+   * collection overhead. Buffer sizes which are less than 256 KB are not allowed, they will be
+   * treated as 256 KB.
+   *
+   * <p>Note that this method does not close neither InputStream nor WriterChannel</p>
+   *
+   * <p>Example of uploading:
+   *
+   * <pre>{@code
+   * Path file = Paths.get("humongous.file");
+   * try (InputStream input = Files.newInputStream(file); WriteChannel writer = storage.writer(blobInfo)) {
+   *     Blob.upload(input, writer, 150 * 1024 * 1024);
+   * }
+   *
+   * @param input content to be uploaded
+   * @param writer channel
+   * @param bufferSize size of the buffer to read from input and send over writer
+   */
+  public static void upload(InputStream input, WriteChannel writer, int bufferSize) {
+    bufferSize = Math.max(bufferSize, 262144);
```

Comment:
why 262144? This magic number could use a named constant for clarity.

Reply:
MIN_BUFFER_SIZE constant added